Jobs/Notifications/StatisticsReportMail.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * @file Jobs/Notifications/StatisticsReportNotify.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class StatisticsReportNotify
+ * @ingroup jobs
+ *
+ * @brief Class to send notification about statistical report to editors
+ */
+
+namespace PKP\Jobs\Notifications;
+
+use APP\core\Application;
+use APP\core\Services;
+use APP\facades\Repo;
+use Context;
+use DateTimeImmutable;
+use Illuminate\Bus\Batchable;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Mail;
+use IntlDateFormatter;
+use PKP\mail\mailables\StatisticsReportNotify;
+use PKP\Support\Jobs\BaseJob;
+use SplFileObject;
+
+class StatisticsReportMail extends BaseJob
+{
+    use Batchable;
+
+    protected Collection $userIds;
+    protected int $contextId;
+    protected DateTimeImmutable $dateStart;
+    protected DateTimeImmutable $dateEnd;
+
+    public function __construct(Collection $userIds, int $contextId, DateTimeImmutable $dateStart, DateTimeImmutable $dateEnd)
+    {
+        parent::__construct();
+
+        $this->userIds = $userIds;
+        $this->contextId = $contextId;
+        $this->dateStart = $dateStart;
+        $this->dateEnd = $dateEnd;
+    }
+
+    public function handle()
+    {
+        $contextDao = Application::getContextDAO();
+        $context = $contextDao->getById($this->contextId); /** @var Context $context */
+        $locale = $context->getPrimaryLocale();
+        $template = Repo::emailTemplate()->getByKey($this->contextId, StatisticsReportNotify::getEmailTemplateKey());
+
+        $editorialTrends = Services::get('editorialStats')->getOverview([
+            'contextIds' => [$context->getId()],
+            'dateStart' => $this->dateStart->format('Y-m-d'),
+            'dateEnd' => $this->dateEnd->format('Y-m-d'),
+        ]);
+        $editorialTrendsTotal = Services::get('editorialStats')->getOverview(['contextIds' => [$context->getId()]]);
+        $totalSubmissions = Services::get('editorialStats')->countSubmissionsReceived(['contextIds' => [$context->getId()]]);
+        $formatter = IntlDateFormatter::create($locale, null, null, null, null, 'MMMM');
+        $month = $formatter->format($this->dateStart);
+        $year = $this->dateStart->format('Y');
+        $filePath = $this->createCsvAttachment($editorialTrends, $editorialTrendsTotal, $locale, $month, $year);
+
+        foreach ($this->userIds as $userId) {
+            $user = Repo::user()->get($userId);
+            $mailable = new StatisticsReportNotify($context, $editorialTrends, $totalSubmissions, $month, $year);
```

Comment:
The report is generated outside of the loop or do you mean mailable? If the latter, it should be generated independently for each separate email because of how [Laravel handles addresses](https://laravel.com/docs/9.x/mail#looping-over-recipients).

Reply:
Right, my mistake. I was thinking that the report was generated in that call but the heavy work is done above.

Regarding the recipients, I think we modified that behaviour in the `Recipient` trait:

https://github.com/pkp/pkp-lib/blob/1e395122fe482a89177864dcf24a6bb0599278f1/classes/mail/traits/Recipient.php#L62-L70